### DISCUSSION TITLE
How do I hide a page for a menu?

### DISCUSSION THREAD

**Discussion by design-innovations** [joomlatools/joomlatools-pages#702] [Q&A] [answered]
How do I hide the homepage from the menu? I thought using visible: false would work, but it's still showing up.

**Comment by johanjanssens** [comment, marked as answer]
You can easily filter pages by adding a custom page property. The property name you use can be anything, I tend to use `visible`for this.  

For example to hide pages from your menus you can add a `visible` property. These are the steps required to set this up:

### 1. Add the custom page property to your config

In `/joomlatools-pages/config.php`add

```php
return array(
    'page' => [
        'visible'   => true,  
    ],
);
```

This defines a custom page property called `visible` that will be added to each page and set to `true` by default. 

### 2. Filter your pages collection for your menu

```php
<? $items = collection('pages', ['folder' => $folder ?? '.', 'level' => $level ?? null,  'recurse' => 'true', 'filter' => ['visible' => 'true']])  ?>
```

Filters your pages using the `visible`property, only adding pages which are visible. For a menu snippet check: https://github.com/joomlatools/joomlatools-pages/discussions/601

### 3. Exclude pages from your menu

You can now easily hide a page from your menu by adding: `visible: false` in the frontmatter of the page

```yaml
Title: Foo
visible: false
```

Thats it! Happy coding.
